File: lib/utils/item_model.dart
class ItemModel {
  final int categoryID;
  final int itemID;
  final String itemName;
  final double itemPrice;
  final String description;
  final String photoPath;

  ItemModel({
    required this.categoryID,
    required this.itemID,
    required this.itemName,
    required this.itemPrice,
    required this.description,
    required this.photoPath,
  });

  factory ItemModel.fromJson(Map<String, dynamic> json) {
    return ItemModel(
      categoryID: json['categoryID'] as int,
      itemID: json['itemID'] as int,
      itemName: json['itemName'] as String,
      itemPrice: (json['itemPrice'] as num).toDouble(),
      description: json['description'] as String,
      photoPath: json['photoPath'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "categoryID": categoryID,
      "itemID": itemID,
      "itemName": itemName,
      "itemPrice": itemPrice,
      "description": description,
      "photoPath": photoPath,
    };
  }

  @override
  bool operator ==(Object other) =>
      other is ItemModel &&
      other.categoryID == categoryID &&
      other.itemID == itemID;

  @override
  int get hashCode => categoryID.hashCode ^ itemID.hashCode;
}

File: admin/backend/src/models/ItemModel.ts
import { Schema, model } from "mongoose";
import { Item } from "../../../frontend/src/utils/types.js";

export const itemSchema = new Schema<Item>({
    categoryID: {
        type: Number,
        required: true,
    },
    itemID: {
        type: Number,
        required: true,
    },
    itemName: {
        type: String,
        required: true,
    },
    itemPrice: {
        type: Number,
        required: true,
    },
    description: {
        type: String,
        required: true,
        default: '',
    },
    photoPath: {
        type: String,
        required: true,
    }
}, {
    autoCreate: false
});

export const ItemModel = model<Item>('Item', itemSchema);
